model/riscv_types_kext.sail

```diff
@@ -0,0 +1,382 @@
+/*
+ * This file contains types, mappings and functions used across the
+ * cryptography extension instructions.
+ *
+ * This file must be included in the model build whatever the value of XLEN.
+ */
+
+/* 
+ * Cryptography extension shared / utility functions
+ * ----------------------------------------------------------------------
+ */
+
+val ror32 : (bits(32), int) -> bits(32)
+function ror32(x, y) = {
+  (x>>to_bits(5,y)) | (x<<to_bits(5,32-y))
```

Comment:
But I believe the following is a nicer way to support rotates (it compiles, possible I messed something up in the calculations though):
```diff
diff --git a/model/prelude.sail b/model/prelude.sail
index e4f9943..52435c8 100644
--- a/model/prelude.sail
+++ b/model/prelude.sail
@@ -180,6 +180,28 @@ function shift_right_arith32 (v : bits(32), shift : bits(5)) -> bits(32) =
     let v64 : bits(64) = EXTS(v) in
     (v64 >> shift)[31..0]
 
+infix 7 >>>
+infix 7 <<<
+
+val rotate_bits_right : forall 'n 'm, 'm >= 0. (bits('n), bits('m)) -> bits('n)
+function rotate_bits_right (v, n) =
+    (v >> n) | (v << (to_bits(length(n), length(v)) - n))
+
+val rotate_bits_left : forall 'n 'm, 'm >= 0. (bits('n), bits('m)) -> bits('n)
+function rotate_bits_left (v, n) =
+    (v << n) | (v >> (to_bits(length(n), length(v)) - n))
+
+val rotater : forall 'm 'n, 'm >= 'n >= 0. (bits('m), atom('n)) -> bits('m)
+function rotater (v, n) =
+    (v >> n) | (v << (length(v) - n))
+
+val rotatel : forall 'm 'n, 'm >= 'n >= 0. (bits('m), atom('n)) -> bits('m)
+function rotatel (v, n) =
+    (v << n) | (v >> (length(v) - n))
+
+overload operator >>> = {rotate_bits_right, rotater}
+overload operator <<< = {rotate_bits_left, rotatel}
+
 /* helpers for mappings */
 
 val spc : unit <-> string
```
That way it's all polymorphic and also supports operator overloading to mirror shifts. It's also a standard enough function that it just belongs in the prelude rather than something specific to K (I assume bitmanip will want it, for example).